graph/coloring/bfs.go

```diff
@@ -0,0 +1,53 @@
+package coloring
+
+import "container/list"
+
+// ColoringUsingBFS will return the Color of each vertex and the
+// total number of different colors used, using BFS
+func (g *Graph) ColoringUsingBFS() (map[int]Color, int) {
+	// Initially all vertices will have same color
+	vertexColors := make(map[int]Color, g.vertices)
+	for i := 0; i < g.vertices; i++ {
+		vertexColors[i] = 1
+	}
+
+	visited := make(map[int]struct{})
+	// Run BFS from each non-visited vertex
+	for i := 0; i < g.vertices; i++ {
+		if _, ok := visited[i]; ok {
+			continue
+		}
+		visited[i] = struct{}{}
+
+		queue := list.New()
```

Comment:
Could it not be possible to use `TheAlgorithms/Go`repo's implementation of a queue? I think its almost same, or am I missing something?

Reply:
Yeah, we can, I used the `container/list` because of the reflexes.